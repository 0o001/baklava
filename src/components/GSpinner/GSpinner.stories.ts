// @ts-nocheck
import { storiesOf } from '@storybook/vue';
import GSpinner from './GSpinner.vue';

storiesOf('GSpinner', module)
  .addParameters({ component: GSpinner })
  .lokiSkip('Base', () => ({
    components: { GSpinner },
    template: `<GSpinner/>`,
  }), {
    notes: 'A very simple example of addon notes',
  })
  .lokiSkip('Small', () => ({
    components: { GSpinner },
    template: `<GSpinner variant="small" />`,
  }), {
    notes: 'A very simple example of addon notes',
  })
  .lokiSkip('Extra Small', () => ({
    components: { GSpinner },
    template: `<GSpinner variant="extra-small" />`,
  }), {
    notes: 'A very simple example of addon notes',
  });
