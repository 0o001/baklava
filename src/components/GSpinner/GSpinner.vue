<template>
  <div :class="classNames">
    <div />
    <div />
    <div />
    <div />
  </div>
</template>

<script lang="ts">

import $_ from '../../utils/dashUtil.ts';

export default {
  name: 'Spinner',
  props: {
    variant: {
      type: String,
      default: 'medium',
    },
  },
  computed: {
    classNames () {
      return {
        'g-spinner': true,
        [$_(this.variant)]: true,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
.g-spinner {
  display: inline-block;
  position: relative;

  &.-medium {
    width: 24px;
    height: 24px;

    div {
      width: 24px;
      height: 24px;
    }
  }

  &.-small {
    width: 16px;
    height: 16px;

    div {
      width: 16px;
      height: 16px;
      margin: 3px;
    }
  }

  &.-extra-small {
    width: 10px;
    height: 10px;

    div {
      width: 10px;
      height: 10px;
    }
  }
}

.g-spinner div {
  display: block;
  position: absolute;
  width: 20px;
  height: 20px;
  margin: 3px;
  border: 2px solid #f27a1a;
  border-radius: 50%;
  animation: g-spinner 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  border-color: #f27a1a transparent transparent transparent;
}

.g-spinner div:nth-child(1) {
  animation-delay: -0.45s;
}

.g-spinner div:nth-child(2) {
  animation-delay: -0.3s;
}

.g-spinner div:nth-child(3) {
  animation-delay: -0.15s;
}

@keyframes g-spinner {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
</style>
