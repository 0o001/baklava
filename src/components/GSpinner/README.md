# Spinner

## Props

| Name      | Type     | Description | Default    |
| --------- | -------- | ----------- | ---------- |
| `variant` | `String` |             | `"medium"` |

## Computed Properties

| Name         | Type     | Description                 |
| ------------ | -------- | --------------------------- |
| `classNames` | `object` | **Dependencies:** `variant` |

