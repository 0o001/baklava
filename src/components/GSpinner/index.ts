import GSpinner from './GSpinner.vue';

export default GSpinner;
