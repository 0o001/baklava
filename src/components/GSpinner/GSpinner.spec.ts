import { shallowMount } from '@vue/test-utils';
import GSpinner from './';

describe('GSpinner', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallowMount(GSpinner);
  });

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render medium variant as default when variant prop does not set', () => {
    expect(wrapper.vm.classNames).toEqual({ 'g-spinner': true, '-medium': true });
  });


  it('should render with given variant', async () => {
    // given
    wrapper.setProps({
      variant: 'extra-small',
    });

    // when
    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.classNames).toEqual({ 'g-spinner': true, '-extra-small': true });
  });
});
