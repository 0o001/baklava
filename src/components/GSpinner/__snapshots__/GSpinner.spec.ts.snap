// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GSpinner should match snapshot 1`] = `
<div
  class="g-spinner -medium"
>
  <div />
   
  <div />
   
  <div />
   
  <div />
</div>
`;
